src/fidesops/api/v1/endpoints/privacy_request_endpoints.py

```diff
@@ -643,14 +626,40 @@ def resume_with_manual_input(
             detail=f"Cannot resume privacy request '{privacy_request.id}'; no paused collection.",
         )
 
-    validate_manual_input(manual_rows, paused_collection, db)
-    logger.info(
-        f"Caching manual input for privacy request '{privacy_request_id}', collection: '{paused_collection}'"
-    )
-    privacy_request.cache_manual_input(paused_collection, manual_rows)
+    if paused_step != expected_paused_step:
+        raise HTTPException(
+            status_code=HTTP_400_BAD_REQUEST,
+            detail=f"Collection '{paused_collection}' is paused at the {paused_step} step. Pass in manual data instead to "
+            f"'{PRIVACY_REQUEST_MANUAL_ERASURE if paused_step == ActionType.erasure else PRIVACY_REQUEST_MANUAL_INPUT}' to resume.",
+        )
+
+    datasets = DatasetConfig.all(db=db)
+    dataset_graphs = [dataset_config.get_graph() for dataset_config in datasets]
+    dataset_graph = DatasetGraph(*dataset_graphs)
+
+    node: Optional[Node] = dataset_graph.nodes.get(paused_collection)
+    if not node:
+        raise HTTPException(
+            status_code=HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=f"Cannot save manual data. No collection in graph with name: '{paused_collection.value}'.",
+        )
+
+    if paused_step == ActionType.access:
+        validate_manual_input(manual_rows or [], paused_collection, dataset_graph)
+        logger.info(
+            f"Caching manual input for privacy request '{privacy_request_id}', collection: '{paused_collection}'"
+        )
+        privacy_request.cache_manual_input(paused_collection, manual_rows)
+
+    elif paused_step == ActionType.erasure:
+        logger.info(
+            f"Caching manually erased row count for privacy request '{privacy_request_id}', collection: '{paused_collection}'"
+        )
+        privacy_request.cache_manual_erasure_count(paused_collection, manual_count)
```

Comment:
Hm, that's not true, the elif statement will be False if the paused step is not an erasure, so it will not get logged or cached.  

Technically, the only possible values for paused_step are access or erasure, as these are the only values cached by the manual connector.  I was trying to re-use ActionType, but it seems like a path forward could be using a literal type `Literal["access", "erasure"]` to match what is possible from a code perspective.

Reply:
You are correct about the cache. It looked un-intended on on the web browser so I thought it was outside the `elsif`. I pulled it up in my editor and now see it is part of the `elsif`.

Either of your options for ActionType would work. You could also still use ActionType and add a check. Something like:

```py
if paused_step not in (ActionType.access, ActionType.erasure):
    raise ValueError("Unsupported action type")
```